x/auth/keeper/deterministic_test.go

```diff
@@ -118,3 +124,425 @@ func pubkeyGenerator(t *rapid.T) *rapid.Generator[secp256k1.PubKey] {
 		return secp256k1.PubKey{Key: pkBz}
 	})
 }
+
+func (suite *DeterministicTestSuite) runAccountsIterations(prevRes []types.AccountI) {
+	for i := 0; i < 1000; i++ {
+		res, err := suite.queryClient.Accounts(suite.ctx, &types.QueryAccountsRequest{})
+		suite.Require().NoError(err)
+		suite.Require().NotNil(res)
+		suite.Require().NotNil(res.Accounts)
+		suite.Require().Len(res.Accounts, len(prevRes))
+
+		unpackedAccs := make([]types.AccountI, len(res.Accounts))
+		for i := 0; i < len(res.Accounts); i++ {
+			var account types.AccountI
+			err = suite.encCfg.InterfaceRegistry.UnpackAny(res.Accounts[i], &account)
+			suite.Require().NoError(err)
+
+			unpackedAccs[i] = account
+		}
+
+		sort.Slice(unpackedAccs, func(i2, j int) bool {
+			return unpackedAccs[i2].GetAccountNumber() < unpackedAccs[j].GetAccountNumber()
+		})
+
+		sort.Slice(prevRes, func(i2, j int) bool {
+			return prevRes[i2].GetAccountNumber() < prevRes[j].GetAccountNumber()
+		})
```

Comment:
> Why do we need this sort?

Because the insertion to the store is neither FIFO nor FILO.

>  Is it only for the first `prevRes`?

Yeah, I think we can sort only once for the first query I mean when `i=0`, later we can assign the result(unsorted) to the `prevRes`.